tesseract/tesseract_planning/tesseract_motion_planners/include/tesseract_motion_planners/ompl/impl/ompl_freespace_planner.hpp

```diff
@@ -303,21 +305,27 @@ bool OMPLFreespacePlanner<PlannerType, PlannerSettingsType>::setConfiguration(
   if (config_->svc != nullptr)
     simple_setup_->setStateValidityChecker(config_->svc);
 
-  if (config_->collision_check && config_->collision_continuous && config_->mv == nullptr)
-  {
-    ompl::base::MotionValidatorPtr mv =
-        std::make_shared<ContinuousMotionValidator>(simple_setup_->getSpaceInformation(), env, kin_);
-    simple_setup_->getSpaceInformation()->setMotionValidator(std::move(mv));
-  }
-  else if (config_->collision_check && !config_->collision_continuous && config_->mv == nullptr)
+  // Setup motion validation (i.e. collision checking)
+  if (config_->mv != nullptr)
   {
-    ompl::base::MotionValidatorPtr mv =
-        std::make_shared<DiscreteMotionValidator>(simple_setup_->getSpaceInformation(), env, kin_);
-    simple_setup_->getSpaceInformation()->setMotionValidator(std::move(mv));
+    simple_setup_->getSpaceInformation()->setMotionValidator(config_->mv);
   }
-  else if (config_->mv != nullptr)
+  else
   {
-    simple_setup_->getSpaceInformation()->setMotionValidator(config_->mv);
+    if (config_->collision_check)
+    {
+      ompl::base::MotionValidatorPtr mv;
```

Comment:
I originally exposed the motion validator in the config so it is possible to set something else if desired. Is this still possible with these changes?

Reply:
Yes, the behavior of the motion validator in the config the same as before. I just rearranged the logic here because the check of `config_->mv == nullptr` was used in several if statements in a row